### ISSUE TITLE
sgadmin.sh doesn't work when invoked from a symlink

### ISSUE THREAD

**Issue by HT154** [floragunncom/search-guard#604]
I have some slightly odd deployment requirements that require me to run sgadmin.sh from a specific directory. When invoked from the symlink, sgadmin.sh sets the classpath argument to java relative to the symlink:

```bash
DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

This should account for being called from a symlink like so:

```bash
DIR="$( cd "$( dirname "$( realpath "${BASH_SOURCE[0]}" )" )" && pwd )"
```

**Comment by floragunncom** [comment]
can you prepare a pull request for that?

**Comment by HT154** [comment]
I need to request permission from my employer for formal contributions, which could take... an indeterminate amount of time. I'm using a patch file in my deployment to get sgadmin.sh correct, but it could be some time before I can open a PR

**Comment by HT154** [comment]
@floragunncom Was it intentional for https://github.com/floragunncom/search-guard/commit/bf56ecf8e0224238a6e7424acdef83a334a413e2#diff-7715bbac7a963d83a2abdb87c46f1a5b to swallow the usage info printed when sgadmin is invoked with no args?

**Comment by floragunncom** [comment]
Only stderr (not stdout) gets swallowed to /dev/null and the nor arg usage gets printed out on stdout. I just verified this and I see the usage info as expected.

**Comment by floragunncom** [comment]
@HT154 does https://github.com/floragunncom/search-guard/commit/d0ff328358f6c7b086a5797e79402e02c8299f2e looks good for you?

**Comment by floragunncom** [comment]
What about adding `-P` to pwd and test for readlink?

```bash
#!/bin/bash
SCRIPT_PATH="${BASH_SOURCE[0]}"
if ! [ -x "$(command -v realpath)" ]; then
    if [ -L "$SCRIPT_PATH" ]; then
        
        [ -x "$(command -v readlink)" ] || { echo "Not able to resolve symlink. Install realpath or readlink.";exit 1; }
        
        # try readlink (-f not needed because we know its a symlink)
        DIR="$( cd "$( dirname $(readlink "$SCRIPT_PATH") )" && pwd -P)"
    else
        DIR="$( cd "$( dirname "$SCRIPT_PATH" )" && pwd -P)"
    fi
else
    DIR="$( cd "$( dirname "$(realpath "$SCRIPT_PATH")" )" && pwd -P)"
fi
```

**Comment by HT154** [comment]
@floragunncom I think that covers just about every case you could hope to cover! The `pwd -P` probably isn'y necessary for the `realpath` branch, but it can't hurt to be extra sure. Thanks for taking the time for this :)

**Comment by floragunncom** [comment]
Will be released with upcoming v24.1

**Comment by floragunncom** [comment]
fixed with v24.1 (also for hash.sh and the demo installer)
